src/impl_constructors.rs

```diff
@@ -132,18 +132,13 @@ where
     /// use approx::assert_abs_diff_eq;
     /// use ndarray::{Array, arr1};
     ///
-    /// # fn example() -> Option<()> {
```

Comment:
Does this cause a `clippy` failure? Given that the return type is `Option<()>` I wouldn't an expect this to be deemed a `needless_doctest_main`.

Reply:
Lemme double-check why I've done so. It might be for the sake of simplicity for the example, but I agree it wouldn't be the exact scope of this pull request.
Given I couldn't find an explanation in the commit message (I usually leave explanatory messages), going to re-run it.

Reply:
Yep, running fine with the example function. So it was, indeed, a matter of simplicity